test/test_orbit/test_orbit.cpp

```diff
@@ -0,0 +1,372 @@
+#include <stdio.h>
+#include <orb/Orbit.h>
+#include <cstdint>
+#include <limits>
+#include <lin.hpp>
+#include <gnc/constants.hpp>
+#include <gnc/config.hpp>
+
+
+//UTILITY MACROS
+#include <unity.h>
+#include "../custom_assertions.hpp"
+
+#define NANOSECONDS_IN_WEEK (7ULL*24ULL*60ULL*60ULL*1'000'000'000ULL)
+
+//0 or 1 that can't be calculated at compile time.
+#ifdef DESKTOP
+#define notcompiletime (rand()%2)
+#else
+#include <Arduino.h>
+#define notcompiletime (millis()%2)
+#endif
```

Comment:
Is this just to prevent the compiler from doing everything at compile time?

Reply:
Yeah